portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -70,4 +82,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   private String convertToJsonUsingGson(List<String> comments) {
     return new Gson().toJson(comments);
   }
+
+  private int getNumComments(HttpServletRequest request) { 
+    String numCommentsString = request.getParameter("numComments");
+
+    // Convert the input to an int.
+    int numComments;
+    try {
+      numComments = Integer.parseInt(numCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + numCommentsString);
+      return -1;
+    }
+
+    // Check that the input is not negative.
+    if (numComments < 0) {
+      System.err.println("Cannot display negative comments: " + numCommentsString);
+      return -1;
+    }
```

Comment:
This section isn't necessary. Returning a negative value when you can't parse is fine, but if the user sends a negative value, there's no reason this method should check for that. It's up to the method that calls this to decide what to do with the value returned. Remember: helper method.